### ISSUE TITLE
Exception while HyperParameterTuning -> java.lang.IllegalArgumentException: Type must either be DataSetIterator or DataSetIteratorFactory

### ISSUE THREAD

**Issue by rahul-raj** [deeplearning4j/deeplearning4j#5445]
Here is the updated code ->
https://github.com/rahul-raj/Deeplearning4J/blob/master/src/main/java/HyperParamTuning.java

And here is the stacktrace after executing the code:
```
12:01:34.712 [main] INFO  o.d.a.saver.local.FileModelSaver - FileModelSaver saving networks to local directory: resources
12:01:35.919 [main] INFO  o.d.a.o.r.BaseOptimizationRunner - LocalOptimizationRunner: execution started
12:01:37.201 [main] INFO  org.nd4j.linalg.factory.Nd4jBackend - Loaded [CpuBackend] backend
12:01:38.086 [main] INFO  org.nd4j.nativeblas.NativeOpsHolder - Number of threads used for NativeOps: 4
12:01:38.405 [main] INFO  org.nd4j.nativeblas.Nd4jBlas - Number of threads used for BLAS: 4
12:01:38.409 [main] INFO  o.n.l.a.o.e.DefaultOpExecutioner - Backend used: [CPU]; OS: [Windows 10]
12:01:38.409 [main] INFO  o.n.l.a.o.e.DefaultOpExecutioner - Cores: [8]; Memory: [1.8GB];
12:01:38.409 [main] INFO  o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [OPENBLAS]
12:01:38.521 [LocalCandidateExecutor-0] INFO  o.d.nn.multilayer.MultiLayerNetwork - Starting MultiLayerNetwork with WorkspaceModes set to [training: ENABLED; inference: ENABLED], cacheMode set to [NONE]
12:01:38.659 [LocalCandidateExecutor-0] WARN  o.d.a.t.MultiLayerNetworkTaskCreator - Execution failed for task 0
java.lang.IllegalArgumentException: Type must either be DataSetIterator or DataSetIteratorFactory
	at org.deeplearning4j.arbiter.scoring.util.ScoreUtil.getIterator(ScoreUtil.java:68) ~[arbiter-deeplearning4j-1.0.0-beta.jar:na]
	at org.deeplearning4j.arbiter.task.MultiLayerNetworkTaskCreator$DL4JLearningTask.callHelper(MultiLayerNetworkTaskCreator.java:168) [arbiter-deeplearning4j-1.0.0-beta.jar:na]
	at org.deeplearning4j.arbiter.task.MultiLayerNetworkTaskCreator$DL4JLearningTask.call(MultiLayerNetworkTaskCreator.java:114) [arbiter-deeplearning4j-1.0.0-beta.jar:na]
	at org.deeplearning4j.arbiter.task.MultiLayerNetworkTaskCreator$DL4JLearningTask.call(MultiLayerNetworkTaskCreator.java:85) [arbiter-deeplearning4j-1.0.0-beta.jar:na]
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:108) [guava-19.0.jar:na]
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:41) [guava-19.0.jar:na]
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:77) [guava-19.0.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
```

At this moment, I dont have much clue about the issue. I was following this example:
https://github.com/deeplearning4j/dl4j-examples/blob/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/arbiter/BasicHyperparameterOptimizationExample.java

Just replaced few deprecated usages in the above example with required ones and added my own data provider to retrieve CSV data. Let me know if more info required. Thank you!

**Comment by rahul-raj** [comment]
Source code updated now:
https://github.com/rahul-raj/Deeplearning4J/blob/master/src/main/java/HyperParamTuning.java

Currently observing the below exception:
[error.log](https://github.com/deeplearning4j/deeplearning4j/files/2084541/error.log)

Knowing that splitter isn't recognizing the labels, but ultimately it should get them from underlying iterators. I tried to find a way where I could set labels at reader/iterator level. But no luck.

**Comment by rahul-raj** [comment]
#5521 
This is exactly the status of current execution. Hence marking this as closed, will have an eye on #5521
